core/templates/dev/head/editor/RouterServices.js

```diff
@@ -137,34 +155,34 @@ oppia.factory('routerService', [
       },
       navigateToMainTab: function(stateName) {
         _savePendingChanges();
-
         if (_getCurrentStateFromLocationPath() === stateName) {
           return;
         }
 
-        var _actuallyNavigate = function(newStateName) {
-          if (newStateName) {
-            editorContextService.setActiveStateName(newStateName);
-          }
-          $location.path('/gui/' + editorContextService.getActiveStateName());
-          $window.scrollTo(0, 0);
-        };
-
         if (_tabs.active === MAIN_TAB) {
           $('.oppia-editor-cards-container').fadeOut(function() {
-            _actuallyNavigate(stateName);
+            _actuallyNavigate('gui', stateName);
             $rootScope.$apply();
             $timeout(function() {
               $('.oppia-editor-cards-container').fadeIn();
             }, 150);
           });
         } else {
-          _actuallyNavigate(stateName);
+          _actuallyNavigate('gui', stateName);
         }
       },
-      navigateToPreviewTab: function() {
-        _savePendingChanges();
-        $location.path('/preview');
+      navigateToPreviewTab: function(stateName) {
+        if (_tabs.active === PREVIEW_TAB) {
```

Comment:
Hm. If this is the case -- maybe just do nothing? We'll have a reset button later.

Otherwise, you're bringing DOM references into a JS service, which is weird.